### PR TITLE
[lldb] Merge SwiftLanguageRuntimeImpl back into SwiftLanguageRuntime (NFC)

### PR DESCRIPTION
**Pull request by adrian-prantl**
The lazy initialization of the Runtime used to be necessary before the
ReflectionContext was able to initialized lazily. Since we can just
defer initializing ReflectionContext, we can get rid of the extra
indirection.

rdar://139995777

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Review by JDevlieghere — APPROVED** [review]

File: lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeImpl.h [hunk 1]
```diff
@@ -9,7 +9,7 @@
 // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 //===----------------------------------------------------------------------===//
-
+#error cxxcxxx
```
Comment (JDevlieghere) [inline lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeImpl.h]: ???

**Review by JDevlieghere — CHANGES_REQUESTED** [review]

File: lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeImpl.h (hunk 1 above)
Comment (adrian-prantl) [reply lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeImpl.h]: I meant to `git rm` this. Done!

**Review by JDevlieghere — APPROVED** [review]

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test

**Pull request merged**